### PR TITLE
Service objects

### LINKED ISSUE #16: [rewrite] Break up into service objects/resources
I don't like approach of just appending functions to the `IntelOwlClient` struct.

This SDK should be created following the approach given in [this](https://medium.com/@marcus.olsson/writing-a-go-client-for-your-restful-api-c193a2f4998c) blog-post, especially the "Service Objects" heading.

This approach is very similar to how we (at Certego) wrote [pydragonfly](https://github.com/certego/pydragonfly) recently using the [django-rest-client](https://github.com/certego/django-rest-client) framework (also written by us).

### LINKED ISSUE #21: Break down each API endpoint to service objects
Separate the API endpoints to their specific resource and make a service object in separate files. Furthermore, provide optional struct parameters.

- [ ] Separate the endpoints into service objects.
- [ ] Add their optional structs.
- [ ] Integrate context into the methods.

### PR DESCRIPTION
**Pull request by fear-the-reaper**
@eshaan7
For issue #21 
## Work Done
- Added a basic client with an option struct param 
- Added a basic implementation of the tag service object. 
- Provided stubs for Tag resource error implementation.

### CODE REVIEW

**Comment by fear-the-reaper** [comment]
The latest commit added the tag service object and implemented the error interface!

File: gointelowl/tags.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	fmt.Println("Url: " + requestUrl)
+	request, err := http.NewRequest("DELETE", requestUrl, nil)
+	if err != nil {
+		return false, err
+	}
+	successResp, err := tagService.client.makeRequest(ctx, request)
+	if err != nil {
+		return false, err
+	}
+	if successResp.StatusCode == 204 {
```
Comment (sp35) [inline gointelowl/tags.go:136]: We should change this to `http.StatusNoContent`

File: gointelowl/tags.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	request, err := http.NewRequest("GET", requestUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+	successResp, err := tagService.client.makeRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	var tagList []Tag
+	json.Unmarshal(successResp.Data, &tagList)
```
Comment (sp35) [inline gointelowl/tags.go:49]: what if there's an error while unmarshalling the data

File: gointelowl/tags.go (hunk 1 above)
Comment (fear-the-reaper) [reply gointelowl/tags.go:136]: Will do!

File: gointelowl/tags.go (hunk 2 above)
Comment (fear-the-reaper) [reply gointelowl/tags.go:49]: Yes! Handled that.

**Pull request closed without merging**